include/aspect/assemblers.h

```diff
@@ -0,0 +1,106 @@
+/*
+  Copyright (C) 2016 by the authors of the ASPECT code.
```

Comment:
I am ok with merging the files again for now (for this PR), but ultimately if we allow different implementations of these terms we should also split that visibly in different files. `assembly.cc` and `assembly.h` should only contain the not duplicative parts. I will merge assemblers.h and assembly.h and keep an old branch around for future plans.